src/binary_tree_partition.jl

```diff
@@ -41,10 +41,11 @@ Return the root vertex of a directed tree data graph
 end
 
 """
-Check if a data graph is a directed binary tree
+Check if a named graph is a directed binary tree
+  TODO: we may want to move this to `NamedGraphs.jl`
 """
-@traitfn function _is_directed_binary_tree(graph::AbstractDataGraph::IsDirected)
-  if !is_tree(undirected_graph(underlying_graph(graph)))
+@traitfn function _is_directed_binary_tree(graph::AbstractNamedGraph::IsDirected)
```

Comment:
Is this more specifically checking if it is a [rooted directed tree](https://en.wikipedia.org/wiki/Arborescence_(graph_theory)) (also known as an arborescence)? Since of course there are directed trees that are not rooted, i.e. [polytrees](https://en.wikipedia.org/wiki/Polytree). All of this terminology is a bit crazy but I think it would would be best to be as specific as possible so we can understand it when we read it later, so I would name this `_is_rooted_directed_binary_tree` if it also implicitly checks if it is rooted.

Reply:
@mtfishman thanks, yes these make sense. I would like to write it in a better way so refactored these set of functions in #74 , I will update this PR once that is merged